src/Symfony/Component/WebLink/Link.php

```diff
@@ -50,10 +66,11 @@ class Link implements EvolvableLinkInterface
      */
     private array $attributes = [];
 
-    public function __construct(string $rel = null, string $href = '')
+    public function __construct(string|LinkRelation|null $rel = null, string $href = '')
     {
         if (null !== $rel) {
-            $this->rel[$rel] = $rel;
+            $relEnum = is_string($rel) ? LinkRelation::from($rel) : $rel;
```

Comment:
using an enum makes it impossible to use other values for `rel`, which this is extensible (different specs are providing values, and new specs might appear). To me, we should keep using a string here, not an enum. Enums are for closed lists, and this does not look like a closed list.

Reply:
In order to store a string we can do the conversion in the other direction

```
if (null !== $rel) {
            $rel = ($rel::class === LinkRelation::class) ? $rel->value : $rel;
            $this->rel[$rel] = $rel;
        }
```

Reply:
The ::class does not work on a string.
I will replace with !is_string

Reply:
I had an idea this morning.
There is a standard and it's a shame not to use it.

So I propose :
 - add an attribute $customRel which will be a string[]
 - modify the type of $rel which will be a LinkRelation[]
 - modify the methods getRels, withRel and withoutRel take into account this evolution without BC
 - add the method getStandardRel which will return only the standard rel.

I don't know if withStandardRel and withoutStandardRel are relevant.

What do you think about this? I would modify in this case if it suits you.

Translated with www.DeepL.com/Translator (free version)

Reply:
There is not **1** standard definition relations. There is 1 standard defining the `Link` header with a registry to allow **other** standards to extend the list of relations. So this is *standardized* as an opened list, while your usage of an enum implements it as a closed list.
We **already** have a way to expose the standardized relations before your PR (although not the full lists of standards are in the list), by using constants holding a string, while keeping the argument accepting `string` and so supporting an open list.
To me, this looks like a case where your initial goal is to introduce an enum in Symfony rather than trying to solve the problem properly. All complexity goes away when you stop using an enum as part of the solution to represent this open list.

Side note: please don't mark the conversation as resolved when you still expect a response. When the discussion is ongoing, the conversation is definitely not resolved.